apps/dashboard/src/main/java/com/akto/action/DashboardAction.java

```diff
@@ -72,39 +77,92 @@ public String fetchRiskScoreCountMap(){
 
     Set<Integer> deactivatedCollections = UsageMetricCalculator.getDeactivated();
 
-    private long totalIssuesCount = 0;
-    private long oldOpenCount = 0;
+    List<Integer> totalIssuesCountDayWise;
     public String findTotalIssues() {
-        Set<Integer> demoCollections = new HashSet<>();
-        demoCollections.addAll(deactivatedCollections);
-        demoCollections.add(RuntimeListener.LLM_API_COLLECTION_ID);
-        demoCollections.add(RuntimeListener.VULNERABLE_API_COLLECTION_ID);
+        long daysBetween = (endTimeStamp - startTimeStamp) / ONE_DAY_TIMESTAMP;
+        List<Bson> pipeline = new ArrayList<>();
 
-        ApiCollection juiceshopCollection = ApiCollectionsDao.instance.findByName("juice_shop_demo");
-        if (juiceshopCollection != null) demoCollections.add(juiceshopCollection.getId());
+        Bson notIncludedCollections = UsageMetricCalculator.excludeDemosAndDeactivated("_id." + TestingIssuesId.API_KEY_INFO + "." + ApiInfo.ApiInfoKey.API_COLLECTION_ID);
 
+        Bson filters = Filters.and(
+                notIncludedCollections,
+                Filters.gte(TestingRunIssues.CREATION_TIME, startTimeStamp),
+                Filters.lte(TestingRunIssues.CREATION_TIME, endTimeStamp)
+        );
 
-        if (startTimeStamp == 0) startTimeStamp = Context.now() - 24 * 1 * 60 * 60;
-        // totoal issues count = issues that were created before endtimestamp and are either still open or fixed but last updated is after endTimestamp
-        totalIssuesCount = TestingRunIssuesDao.instance.count(
-            Filters.and(
-                Filters.lte(TestingRunIssues.CREATION_TIME, endTimeStamp),
-                Filters.nin("_id.apiInfoKey.apiCollectionId", demoCollections),
-                Filters.eq(TestingRunIssues.TEST_RUN_ISSUES_STATUS,  GlobalEnums.TestRunIssueStatus.OPEN))       
-            );
+        Bson openIssuesMatchStage = Aggregates.match(Filters.and(
+                filters,
+                Filters.in(TestingRunIssues.TEST_RUN_ISSUES_STATUS, GlobalEnums.TestRunIssueStatus.OPEN.name())
+        ));
 
-        // issues that have been created till start timestamp
-        oldOpenCount = TestingRunIssuesDao.instance.count(
-                Filters.and(
-                        Filters.nin("_id.apiInfoKey.apiCollectionId", demoCollections),
-                        Filters.lte(TestingRunIssues.CREATION_TIME, startTimeStamp),
-                        Filters.ne(TestingRunIssues.TEST_RUN_ISSUES_STATUS,  GlobalEnums.TestRunIssueStatus.IGNORED)
-                )
+        pipeline.add(openIssuesMatchStage);
```

Comment:
The filter logic for calculation does not match the previous logic, more specifically, the oldOpenCount is not being adressed in the new code. Is this by design ?

Reply:
before we were using only two points to calculate the difference between old data and new data, but now we are getting all the data between the time ranges, so we do not need to fetch the old data.